pkg/csidriver/driver.go

```diff
@@ -285,20 +291,48 @@ func commonOSRemove(dir, dbg string) error {
 	defer klog.V(4).Infof("commonOSRemove completed delete attempt for dir %q", dir)
 	// we cannot do a os.RemoveAll on the mount point, so we remove all on each file system entity
 	// off of the potential mount point
-	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
-		if info == nil {
-			return nil
-		}
-		// since we do not support mounting on existing content, a dir can only mean a share
-		// has been mounted as a separate dir in our share, so skip
-		if info.IsDir() {
-			return nil
+	dirFile, err := os.Open(dir)
+	if err != nil {
+		return err
+	}
+	dirEntries, err := dirFile.ReadDir(-1)
+	if err != nil {
+		return err
+	}
+	// we have to unlink the key from our configmap/secret first with the atomic writer symlinks
+	for _, dirEntry := range dirEntries {
+		if dirEntry.Type()&os.ModeSymlink != 0 && !strings.HasPrefix(dirEntry.Name(), "..") {
+			// unlink our secret/configmap key
+			err = syscall.Unlink(filepath.Join(dir, dirEntry.Name()))
+			if err != nil {
+				klog.V(0).Infof("commonOSRemove: %s", err.Error())
+				return err
+			}
+			break
 		}
-		fileName := filepath.Join(dir, info.Name())
-		klog.V(4).Infof("commonOSRemove going to delete file %s", fileName)
-		return os.RemoveAll(fileName)
-	})
+	}
+	// we then unlink the symlink from atomic writer that start with ".."
+	for _, dirEntry := range dirEntries {
+		if dirEntry.Type()&os.ModeSymlink != 0 && strings.HasPrefix(dirEntry.Name(), "..") {
+			err = syscall.Unlink(filepath.Join(dir, dirEntry.Name()))
+			if err != nil {
+				klog.V(0).Infof("commonOSRemove: %s", err.Error())
+				return err
+			}
 
+		}
+		// then we delete any non symlinks
+		if dirEntry.Type()&os.ModeSymlink == 0 {
+			fileName := filepath.Join(dir, dirEntry.Name())
+			klog.V(4).Infof("commonOSRemove going to delete file %s", fileName)
+			err = os.RemoveAll(fileName)
+			if err != nil {
+				klog.V(0).Infof("commonOSRemove: %s", err.Error())
+				return err
+			}
```

Comment:
```
if err = os.RemoveAll(fileName); err != nil {
klog.V(0).Infof("commonOSRemove: %s", err.Error())
return err
}
```
If this is going to be loglevel 0, it should probably be more human-friendly text, or maybe you meant it to be loglevel 4 like the others?